src/handlers/misc/Leaderboard.ts

```diff
@@ -12,15 +16,29 @@ export default class LeaderboardHandler extends AbstractHandler {
         const users = await User.find({ order: { points: 'DESC' } })
 
         let result = ''
+        let expired = this.hasExpired();
         for (let i = 0; i < users.length; i++) {
             const currUser = users[i]
-            let discordUser: GuildMember
-            try {
-                discordUser = await msg.guild.members.fetch(currUser.discordId)
-            } catch (e) {
+
+            if(expired){
+
+                let discordUser: GuildMember
+
+                try
+                {
+                    discordUser = await msg.guild.members.fetch(currUser.discordId)
+
+                    const username = await discordUser.user.username
+
+                    await updateName(currUser.discordId, username)
+                }
+
+             catch (e) {
                 continue
+                }
             }
-            let toPush = `${i + 1}. ${discordUser.user.username} ⇆ ${currUser.points}`
+
+            let toPush = `${i + 1}. ${currUser.username} ⇆ ${currUser.points}`
```

Comment:
`currUser.username` isn't guaranteed to exist, so we should think about a fallback to show.

Reply:
Right now, it is. Every user creation initializes the username along with the Discord ID in the database. Existing users will automatically be updated on first run.